pkg/local_workflows/authworkflow/refresh.go

```diff
@@ -0,0 +1,46 @@
+package authworkflow
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/snyk/go-application-framework/pkg/auth"
+	"github.com/snyk/go-application-framework/pkg/workflow"
+	"github.com/spf13/pflag"
+)
+
+const refreshWorkflowName = "auth.refresh"
+
+var refreshWorkflowId = workflow.NewWorkflowIdentifier(refreshWorkflowName)
+
+// InitRefresh registers the oauth token refresh workflow with the engine.
+func InitRefresh(engine workflow.Engine) error {
+	config := pflag.NewFlagSet(refreshWorkflowName, pflag.ExitOnError)
+	config.String(authTypeParameter, "token", "Authentication type (token, oauth)")
+	_, err := engine.Register(refreshWorkflowId, workflow.ConfigurationOptionsFromFlagset(config), refreshEntryPoint)
+	return err
+}
+
+func refreshEntryPoint(invocationCtx workflow.InvocationContext, _ []workflow.Data) (_ []workflow.Data, err error) {
+	config := invocationCtx.GetConfiguration()
+	oauthEnabled := config.GetString(authTypeParameter) == "oauth"
+	if !oauthEnabled { // Only used in OAuth flow
+		const oauthDisabled = "OAuth disabled - cannot refresh token"
+		fmt.Println(oauthDisabled)
+		return nil, errors.New(oauthDisabled)
+	}
+
+	authenticator, ok := invocationCtx.GetNetworkAccess().GetAuthenticator().(*auth.OAuth2Authenticator)
```

Comment:
This should not use the real type. It should just get an authenticator. If you need to use an interface you could e.g. Define a `RefreshableAuth` interface to be used, exported that has a `Refresh` method, but I'd think twice about exporting the concrete type and asserting the type here.